File: app/src/main/java/com/github/premnirmal/ticker/network/historicaldata/HistoricalData.java
package com.github.premnirmal.ticker.network.historicaldata;

import android.os.Parcelable;
import android.os.Parcel;

public class HistoricalData implements Parcelable {

  public Query query;

  public HistoricalData() {
  }

  public HistoricalData(Parcel in) {
    query = in.readParcelable(Query.class.getClassLoader());
  }

  @Override public int describeContents() {
    return 0;
  }

  public static final Creator<HistoricalData> CREATOR = new Creator<HistoricalData>() {
    public HistoricalData createFromParcel(Parcel in) {
      return new HistoricalData(in);
    }

    public HistoricalData[] newArray(int size) {
      return new HistoricalData[size];
    }
  };

  @Override public void writeToParcel(Parcel dest, int flags) {
    dest.writeParcelable(query, flags);
  }
}

File: app/src/main/kotlin/com/github/premnirmal/ticker/network/data/historicaldata/HistoricalData.kt
package com.github.premnirmal.ticker.network.data.historicaldata

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

/**
 * Created by premnirmal on 3/30/17.
 */
class HistoricalData : Parcelable {

  var query: Query? = null

  constructor()

  constructor(`in`: Parcel) {
    query = `in`.readParcelable<Query>(Query::class.java.classLoader)
  }

  override fun describeContents(): Int {
    return 0
  }

  override fun writeToParcel(dest: Parcel, flags: Int) {
    dest.writeParcelable(query, flags)
  }

  companion object {

    val CREATOR: Creator<HistoricalData> = object : Creator<HistoricalData> {
      override fun createFromParcel(`in`: Parcel): HistoricalData {
        return HistoricalData(`in`)
      }

      override fun newArray(size: Int): Array<HistoricalData> {
        return newArray(size)
      }
    }
  }
}